src/components/SUE/report/SueReportLocation.tsx

```diff
@@ -187,53 +250,19 @@ export const SueReportLocation = ({
           locations={locations}
           mapCenterPosition={mapCenterPosition}
           mapStyle={styles.map}
-          onMyLocationButtonPress={() =>
-            Alert.alert(texts.sue.report.alerts.hint, texts.sue.report.alerts.myLocation, [
-              {
-                text: texts.sue.report.alerts.no
-              },
-              {
-                text: texts.sue.report.alerts.yes,
-                onPress: async () => {
-                  const location = position || lastKnownPosition;
-
-                  if (location) {
-                    setUpdatedRegion(true);
-                    setUpdateRegionFromImage(false);
-                    setSelectedPosition(location.coords);
-
-                    try {
-                      await handleGeocode(location.coords);
-                    } catch (error) {
-                      setSelectedPosition(undefined);
-                      Alert.alert(texts.sue.report.alerts.hint, error.message);
-                    }
-                  }
-                }
-              }
-            ])
-          }
-          onMapPress={async ({ nativeEvent }) => {
-            if (
-              nativeEvent.action !== 'marker-press' &&
-              nativeEvent.action !== 'callout-inside-press'
-            ) {
-              setUpdatedRegion(false);
-              setUpdateRegionFromImage(false);
-              setSelectedPosition(nativeEvent.coordinate);
-
-              try {
-                await handleGeocode(nativeEvent.coordinate);
-              } catch (error) {
-                setSelectedPosition(undefined);
-                Alert.alert(texts.sue.report.alerts.hint, error.message);
-              }
-            }
-          }}
+          onMyLocationButtonPress={onMyLocationButtonPress}
+          onMapPress={onMapPress}
           onMaximizeButtonPress={() =>
-            navigation.navigate(ScreenName.MapView, {
+            navigation.navigate(ScreenName.SueReportMapView, {
               calloutTextEnabled: true,
-              locations
+              currentPosition: position || lastKnownPosition,
```

Comment:
we could use `currentPosition` here as well, if not defined inside the method but right in the component, right?

Reply:
Yes 🙌

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/pull/1140/commits/2cb8a45bc25d8174f8fa77cde39f44c1f838be48